src/snowflake/connector/s3_storage_client.py

```diff
@@ -101,6 +101,14 @@ def __init__(
                     f"https://{self.s3location.bucket_name}.s3.amazonaws.com"
                 )
 
+    @staticmethod
+    def _url_quote(url: str) -> str:
+        """Per Python document at https://docs.python.org/3/library/urllib.parse.html:
+        Changed in version 3.7: Moved from RFC 2396 to RFC 3986 for quoting URL strings.
+        “~” is now included in the set of unreserved characters.
+        This fix for SNOW-480963 is not needed on Python 3.7 and above"""
```

Comment:
Your doc-string format is wrong, please see https://google.github.io/styleguide/pyguide.html#383-functions-and-methods